mlflow/store/sqlalchemy_store.py

```diff
@@ -465,9 +465,6 @@ def set_tag(self, run_id, tag):
 
     def _search_runs(self, experiment_ids, filter_string, run_view_type, max_results, order_by,
                      page_token):
-        if page_token:
-            raise MlflowException("SQLAlchemy-backed tracking stores do not yet support pagination"
-                                  "tokens.")
         # TODO: push search query into backend database layer
         if max_results > SEARCH_MAX_RESULTS_THRESHOLD:
```

Comment:
So what does `max_results` mean now? Does it refer to the total number of results wanted (even if it has to be paginated), or does it refer to the number of results desired on the next "page"?

Reply:
I believe it has always referred to the total number of results desired (given some maximum imposed by the server).

Reply:
I think if I'm reading the code right,  in the `paginate` method it returns `max_results` number of runs per "page" until the offset is greater than the total number of runs available. Can you check this?

So for example, if you said `max_results` = 5, and there are 50 runs in total, it will continue sending page tokens back with 5 runs until you've requested 10 pages.

Reply:
Yes, this is the correct behavior, right? The client chooses the page size, but cannot choose an arbitrarily large size (as imposed by the server).

Reply:
Ah, ok. I figured that the UI default would be to show up to 1000 runs per "load more", even if the user queried with `max_results=2500`. But what actually will happen is that the server will send all 2500 runs back in one page, unless the frontend dynamically changes the `max_results` to be the desired page size (currently 1000). (cc @Zangr)

I think with the auto-pagination with pandas and spark datasources, `max_results` will refer to the maximum number of results the user wants in the dataframe, which may be over the server limit for one page. But the functions will split it into multiple smaller pages and combine them.

Reply:
`max_results=1000` defaulted in proto. We can absolutely reduce that with pagination and more importantly `Load More...` implemented.